analyzer/evmtokenbalances/evm_token_balances.go

```diff
@@ -146,25 +155,57 @@ func (m Main) getStaleTokenBalances(ctx context.Context, limit int) ([]*StaleTok
 }
 
 func (m Main) processStaleTokenBalance(ctx context.Context, batch *storage.QueryBatch, staleTokenBalance *StaleTokenBalance) error {
-	m.logger.Info("downloading", "stale_token_balance", staleTokenBalance)
-	// todo: assert that token addr and account addr contexts are secp256k1
-	tokenEthAddr, err := client.EVMEthAddrFromPreimage(staleTokenBalance.TokenAddrContextIdentifier, staleTokenBalance.TokenAddrContextVersion, staleTokenBalance.TokenAddrData)
-	if err != nil {
-		return fmt.Errorf("token address: %w", err)
-	}
 	accountEthAddr, err := client.EVMEthAddrFromPreimage(staleTokenBalance.AccountAddrContextIdentifier, staleTokenBalance.AccountAddrContextVersion, staleTokenBalance.AccountAddrData)
 	if err != nil {
 		return fmt.Errorf("account address: %w", err)
 	}
-	if staleTokenBalance.Type != nil {
-		balanceData, err := runtime.EVMDownloadTokenBalance(
+	switch staleTokenBalance.Type {
+	case evm.EVMTokenTypeUnsupported:
+		// Do nothing; we'll just mark this token as processed so we remove it from the queue.
+	case evm.EVMTokenTypeNative:
+		// Query native balance.
+		addr := nodeapi.Address{}
+		if err := addr.UnmarshalText([]byte(staleTokenBalance.AccountAddr)); err != nil {
+			m.logger.Error("invalid account address bech32 '%s': %w", staleTokenBalance.AccountAddr, err)
+			break
+		}
+		balance, err := m.source.GetNativeBalance(ctx, staleTokenBalance.DownloadRound, addr)
+		if err != nil || balance == nil {
```

Comment:
Is the additional balance nil check redundant? I'm not used to seeing such a pattern in go code, unless the function returning nil is an actual expected case.

Reply:
It's redundant in that we don't expect `GetNativeBalance()` to ever return `nil, nil`. I removed the check now for consistency. Man, Go's cavalier approach to possible nilness ... :rage: ... like Java in the 90s taught us nothing.

Reply:
:shrug: